Sources/Swift/Integrations/SessionReplay/SentryBreadcrumbReplayConverter.swift

```diff
@@ -0,0 +1,92 @@
+        } else if breadcrumb.type == "navigation" {
+            return navigationBreadcrumb(breadcrumb)
+        } else if breadcrumb.category == "touch" {
+            return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: "ui.tap", message: breadcrumb.message)
+        } else if breadcrumb.type == "connectivity" && breadcrumb.category == "device.connectivity" {
+            guard let networkType = breadcrumb.data?["connectivity"] as? String, !networkType.isEmpty  else { return nil }
+            return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: "device.connectivity", data: ["state": networkType])
+        } else if let action = breadcrumb.data?["action"] as? String, action == "BATTERY_STATE_CHANGE" {
+            var data = breadcrumb.data?.filter({ item in item.key == "level" || item.key == "plugged" }) ?? [:]
+            
+            data["charging"] = data["plugged"]
+            data["plugged"] = nil
+            
+            return SentryRRWebBreadcrumbEvent(timestamp: timestamp,
+                                              category: "device.battery",
+                                              data: data)
+        }
+        
+        let level = getLevel(breadcrumb: breadcrumb)
+        return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: breadcrumb.category, message: breadcrumb.message, level: level, data: breadcrumb.data)
+    }
+    
+    private func navigationBreadcrumb(_ breadcrumb: Breadcrumb) -> SentryRRWebBreadcrumbEvent? {
+        guard let timestamp = breadcrumb.timestamp else { return nil }
+        
+        if breadcrumb.category == "app.lifecycle" {
+            guard let state = breadcrumb.data?["state"] else { return nil }
+            return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: "app.\(state)")
+        } else if let position = breadcrumb.data?["position"] as? String, breadcrumb.category == "device.orientation" {
+            return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: "device.orientation", data: ["position": position])
+        } else {
+            if let to = breadcrumb.data?["screen"] as? String {
+                return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: "navigation", message: to, data: ["to": to])
+            } else {
+                return nil
+            }
+        }
+    }
+    
+    private func networkSpan(_ breadcrumb: Breadcrumb) -> SentryRRWebSpanEvent? {
+        guard let timestamp = breadcrumb.timestamp,
+              let description = breadcrumb.data?["url"] as? String,
+              let startTimestamp = breadcrumb.data?["request_start"] as? Date
+        else { return nil }
+        var data = [String: Any]()
+        
+        breadcrumb.data?.forEach({
+            guard supportedNetworkData.contains($0.key) else { return }
+            let newKey = $0.key == "response_body_size" ? "bodySize" : $0.key.replacingOccurrences(of: "http.", with: "")
```

Comment:
> $0.key == "response_body_size" 

`h`: I think this condition will never be met, because `supportedNetworkData` does not contain `response_body_size`/`request_body_size`. You should probably add these keys to the list

Reply:
and I think in general this logic is slightly off - basically the keys should be `data['requestBodySize']` or `data['responseBodySize']` - I hope it makes more sense this way

Reply:
But If I got it right from java, you're using `bodySize`
 ```Java
 val breadcrumbData = mutableMapOf<String, Any?>()
            for ((key, value) in breadcrumb.data) {
                if (key in supportedNetworkData) {
                    breadcrumbData[
                        key
                            .replace("content_length", "body_size")
                            .substringAfter(".")
                            .snakeToCamelCase()
                    ] = value
                }
            }
```
Not sure If I got your comment.

Reply:
yeah, I guess I wasn't clear enough - so you have this line:
```
guard supportedNetworkData.contains($0.key) else { return }
```
but because `supportedNetworkData` doesn't contain `response_body_size`/`request_body_size` in it, this will always early-return, so you will never get to the next line for those keys. So you have to update `supportedNetworkData` with those keys.

to the second point - yes we use `bodySize` but here we actually replace a `content_length` substring with `body_size`, so `response_content_length` transforms to `response_body_size` and then we camelCase it to `responseBodySize`. I hope it makes it more clear